### PR TITLE
Mentorlist update(without connected people)

### PR DESCRIPTION
**Pull request by SarahhhC**
Getting Mentorlist updated.
We exclude connected people in mentorlist.(status: pending, accept, reject all)

### CODE REVIEW

File: src/server/controllers/users.controller.js [hunk 1]
```diff
@@ -36,18 +37,49 @@ export function getAll(req, res, next) {
     });
 }
 
-// Get all user list except logged in user
 export function getMentorList(req, res, next) {
-  User.find({ _id: { $ne: req.user._id }, mentorMode: { $ne: false } })
-    .sort({ stamp_login: -1 }).exec()
-    .then(mentorList => {
-      res.status(200).json(mentorList);
+  let exceptList = [];
```
Comment (festinalente91) [inline src/server/controllers/users.controller.js]: use const

File: src/server/controllers/users.controller.js [hunk 2]
```diff
@@ -36,18 +37,49 @@ export function getAll(req, res, next) {
...
+  let match = { mentor_id: ObjectId(req.user._id) };
+  let project = { mentee_id: 1 };
```
Comment (ifndefdeadmau5) [inline src/server/controllers/users.controller.js]: Could you explain me about 'project' a little? Not sure what does that mean

File: src/server/controllers/users.controller.js (hunk 1 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Thanks!

File: src/server/controllers/users.controller.js [hunk 3]
```diff
@@ -36,18 +37,49 @@ export function getAll(req, res, next) {
...
+  findConnection(match, project, 'mentee_id')
+    .then((menteelist) => {
+      menteelist.forEach(user => exceptList.push(user.mentee_id));
```
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Sure modified!

File: src/server/controllers/users.controller.js (hunk 2 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: It's for projection.
We can get specific fields from aggregate result and What I want is mentee_id!
The result is like 
```
[ { _id: 5845058577fd976cb25ef0b0,
    mentee_id: 584123581282dd570c7cf093,
    list: [ [Object] ] } ]
```
Cause default field id _id.

Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: In list, you can get detail information of user collection(of foreignField id).

Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: Yes, it's one of mongoDB operation. Thanks for the review!

File: src/server/controllers/users.controller.js [hunk 4]
```diff
@@ -36,18 +37,51 @@ export function getAll(req, res, next) {
...
 }
 
-// Get all user list except logged in user
 export function getMentorList(req, res, next) {
-  User.find({ _id: { $ne: req.user._id }, mentorMode: { $ne: false } })
-    .sort({ stamp_login: -1 }).exec()
-    .then(mentorList => {
-      res.status(200).json(mentorList);
+  const exceptList = [];
+  const project = {
```
Comment (MechanicKim) [inline src/server/controllers/users.controller.js]: this project object is necessary?
I can't find this uses of the object.

Comment (sv-yoda) [inline src/server/controllers/users.controller.js]: This used in line 47, 51!

**Comment by MechanicKim** [comment]
LGTM

**Pull request merged**